src/main/java/EventsAndCommands/CommandLoader.java

```diff
@@ -0,0 +1,222 @@
+            .collect(Collectors.toSet());
+            
+        List<ListenerAdapter> instances = new ArrayList<>();
+
+        for (var cmd : cmds) {
+            List<Constructor<?>> ctors = Arrays.asList(cmd.getConstructors());
+            for (var ctor : ctors) {
+                if (hasAcceptableParams(ctor)) {
+                    try {
+                        instances.add(injectListenerAdaptor(ctor));
+                    } catch (InvocationTargetException e){
+                        e.getCause().printStackTrace();
+                    } catch (Exception e) {
+                        // TODO: Log
+                        System.err.println(e.getMessage());
+                        System.err.println("Could not load " + cmd.getName());
+                    }
+                } else {
+                    System.err.println("No suitable constructor for " + cmd.getName());
+                }
+            }
+        }
+
+        setLoadedListeners(instances);
+    }
+
+    /**
+     * 
+     * @param ctors
+     * @return
+     */
+    public ListenerAdapter injectListenerAdaptor(Constructor<?> ctor) throws Exception{
+        Class<?>[] paramTypes = ctor.getParameterTypes();
+        Object[] parameters = new Object[paramTypes.length];
+
+        for (int i = 0; i < paramTypes.length; i++) {
+            Class<?> type = paramTypes[i];
+            if (type.equals(db.getClass())) {
+                parameters[i] = db;
+            } else if (type.equals(config.getClass())) {
+                parameters[i] = config;
+            } else if (type.equals(waiter.getClass())) {
+                parameters[i] = waiter;
+            }
+        }
+
+        return (ListenerAdapter) ctor.newInstance(parameters);
+    }   
+    
+    public Command injectCommand(Constructor<?> ctor) throws Exception{
+        Class<?>[] paramTypes = ctor.getParameterTypes();
+        Object[] parameters = new Object[paramTypes.length];
+
+        for (int i = 0; i < paramTypes.length; i++) {
+            Class<?> type = paramTypes[i];
+            if (type.equals(db.getClass())) {
+                parameters[i] = db;
+            } else if (type.equals(config.getClass())) {
+                parameters[i] = config;
+            } else if (type.equals(waiter.getClass())) {
+                parameters[i] = waiter;
+            }
+        }
+
+        return (Command) ctor.newInstance(parameters);
+    }
+
+    public boolean hasAcceptableParams(Constructor<?> ctor){
+        final Set<Class<?>> acceptableParams = new HashSet<>();
+        acceptableParams.add(MongoDatabase.class);
+        acceptableParams.add(Properties.class);
+        acceptableParams.add(EventWaiter.class);
+
+        return acceptableParams.containsAll(Arrays.asList(ctor.getParameterTypes()));
+    }
+
+    /**
+     * @return the eventHandlers
+     */
+    public List<Command> getLoadedCommands() {
+        return loadedCommands;
```

Comment:
You likely want to return an unmodifiable wrapper (Collections.unmodifiableList) **if** modifications to your inner `loadeCommands` are not wanted.